packages/connect-core/src/transform-stream-story.spec.ts

```diff
@@ -0,0 +1,324 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  createReadableByteStream,
+  createReadableStream,
+  node16WhatwgStreamPolyfill,
+  readAll,
+  readAllBytes,
+} from "./whatwg-stream-helper.spec.js";
+import type { Serialization } from "./serialization.js";
+import {
+  transformCompress,
+  transformDecompress,
+  transformJoin,
+  transformParseWithEnd,
+  transformSerializeWithEnd,
+  transformSplit,
+} from "./transform-stream.js";
+import type { Compression } from "./compression.js";
+import { encodeEnvelopes } from "./envelope.js";
+import { ConnectError } from "./connect-error.js";
+import { Code } from "./code.js";
+
+node16WhatwgStreamPolyfill();
+
+// These tests aim to model the usage of transform streams in clients and servers.
+// Note that the tests were written as a proof of concept, and the coverage is
+// incomplete (cancellation, backpressure, error handling, etc.).
+describe("full story", function () {
+  const readMaxBytes = 0xffffffff; // zlib caps maxOutputLength at this value
```

Comment:
Is it worth reading the max bytes value from here?:  packages/connect-node/src/private/validate-read-max-bytes-option.ts

Reply:
We can't, this is in connect-core.